src/overlay.jsx

```diff
@@ -106,7 +106,11 @@ const Overlay = React.createClass({
 
   _allowScrolling() {
     const body = document.getElementsByTagName('body')[0];
-    body.style.overflow = this._originalBodyOverflow || '';
+    let currentBodyOverflow = body.style.overflow;
+
+    if (currentBodyOverflow !== '') {
+      body.style.overflow = this._originalBodyOverflow || '';
```

Comment:
That's a hack and won't work for people how are applying an overflow property to their body (`!== ''`).
I must admit, the logic is broken. What about this instead:
```diff
  },

   componentDidMount() {
-    this._originalBodyOverflow = document.getElementsByTagName('body')[0].style.overflow;
     if (this.props.show) {
       this._applyAutoLockScrolling(this.props);
     }
@@ -47,7 +46,9 @@ const Overlay = React.createClass({
   },

   componentWillUnmount() {
-    this._allowScrolling();
+    if (this.props.show === true) {
+      this._allowScrolling();
+    }
   },

   _originalBodyOverflow: '',
@@ -101,16 +102,15 @@ const Overlay = React.createClass({

   _preventScrolling() {
     const body = document.getElementsByTagName('body')[0];
+    this._originalBodyOverflow = body.style.overflow;
+
     body.style.overflow = 'hidden';
   },
   _allowScrolling() {
     const body = document.getElementsByTagName('body')[0];
-    let currentBodyOverflow = body.style.overflow;

-    if (currentBodyOverflow !== '') {
-      body.style.overflow = this._originalBodyOverflow || '';
-    }
+    body.style.overflow = this._originalBodyOverflow || '';
   },

   render() {
```